lib/spack/docs/getting_started.rst

```diff
@@ -997,6 +997,83 @@ to add the following to ``packages.yaml``:
+link against libglvnd instead of the underlying implementation.  Environment
+variables set at run time govern the process by which libglvnd loads the
+underlying implementation and dispatches calls to it.  See `this
+<https://github.com/NVIDIA/libglvnd/issues/177#issuecomment-496562769>` comment
+for details on loading a specific GLX implementation and `this
+<https://github.com/NVIDIA/libglvnd/blob/master/src/EGL/icd_enumeration.md>`
+page for information about EGL ICD enumeration.
+
+This codependency between libglvnd and the underlying implementation is modeled
+in Spack with two packages for libglvnd: libglvnd, which provides libglvnd
+proper; and libglvnd-fe, a bundle package that depends on libglvnd and an
+implementation.  Implementations that work through libglvnd are no longer
+providers for graphics virtual dependencies, like "gl" or "glx", but instead
+provide libglvnd versions of these dependencies ("libglvnd-be-gl",
+"libglvnd-be-glx", etc.).  The libglvnd-fe package depends on these
+"libglvnd-be-..." virtual packages, which provide the actual implementation.
+It also depends on libglvnd, itself, and exposes its libraries to downstream
+applications.  For correct operation, the Spack package for the underlying
+implementation has to set the runtime environment to ensure that it is loaded
+when an application linked against libglvnd runs.  This last detail is
+important for users who want to set up an external OpenGL implementation that
+requires libglvnd to work.  This setup requires modifying the ``packages``
+configuration so that the necessary environment variables are set.
+
+.. code-block:: yaml
+
+    packages:
+      opengl:
+        externals:
+        - spec: opengl@4.6+glx+egl+glvnd
+          prefix: /does/not/exist
+          extra_attributes: 
+            glvnd:
+              glx: 'glx_driver_name'
+              egl: '/path/to/egl/vendor'
+        buildable: false
+        variants: +egl+glx+glvnd
+      all:
+        providers:
+          libglvnd-be-gl: [opengl]
+          libglvnd-be-glx: [opengl]
+          libglvnd-be-egl: [opengl]
+
+One final detail about the above example is that it avoids setting the true
+root of the external OpenGL implementation, instead opting to set it to a path
+that is not expected to exist on the system.  This is done for two reasons.
+First, Spack would add directories under this root to environment variables
+that would affect the process of building and installing other packages, such
+as ``PATH`` and ``PKG_CONFIG_PATH``.  These additions may potentially prevent
+those packages from installing successfully, and this risk is especially great
+for paths that house many libraries and applications, like ``/usr``.  Second,
+providing the true root of the external implementation in the ``packages``
+configuration is not necessary because libglvnd need only the environment
+variables set above in the ``packages`` configuration to determine what OpenGL
```

Comment:
It wasn't clear to me that the configuration sets environment variables, reading through the packages themselves I see they set environment variables based on glvnd[glx/egl] - IMO it would be helpful to rephrase this.

Reply:
I added an extra section below the YAML block that explains the two variables set the environment variables and that they should be changed before using.